### ISSUE TITLE
CSD wrapper support

### ISSUE THREAD

**Issue by CecilHarvey** [openconnect/openconnect-gui#37]
as my company's internal network use Cisco Secure Desktop (which is stupid however it's not my position to change it :( ), it would be nice if openconnect-gui can support CSD wrapper functionality, which is supported by the commandline version of openconnect.

**Comment by foal** [comment]
:+1:

**Comment by rvilskerrpi** [comment]
+1 Would be nice!

**Comment by joakim-tjernlund** [comment]
+1 Yes please. Our company also uses CSD

PS.
    Would be nice if the CMAKE build files were fixed for Linux, after some wild hacking it built and
started just fine.

**Comment by horar** [comment]
(I'm just fixing CMake for snapshot build i.e. to be able make a build without cloning whole repo...)

**Comment by joakim-tjernlund** [comment]
Is their a difference in CMake for building snapshot resp. a git clone?

**Comment by horar** [comment]
yes, in 1.4.0 the build from snapshot will fail, because CMake build script get version numbers and other build identifiers (e.g. hash etc.) from git repository - that allow easy identify the source code state;
(building from snapshot will be supported in in short time from version 1.4.1)

**Comment by joakim-tjernlund** [comment]
Ahh, can I download any snapshoot(SHA)?
Or how do I generate build ids from within git?

**Comment by horar** [comment]
snapshot won't work now, just clone the repo or wait few days

**Comment by joakim-tjernlund** [comment]
So I cloned(develop branch) and tried to build but there still are build problems on Linux:
cmake .
-- The CXX compiler identification is GNU 4.9.3
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:45 (get_target_property):
  get_target_property() called with non-existent target
  "Qt5::QWindowsIntegrationPlugin".

Removing the Qt5::QWindowsIntegrationPlugin test gets me a bit longer:

/home/jocke/openconnect-gui/src/vpninfo.cpp:428:17: error: #elif with no expression
 #elif Q_OS_LINUX
                 ^
/home/jocke/openconnect-gui/src/vpninfo.cpp:431:2: error: #error Define OS string of other platf
 #error Define OS string of other platforms...
  ^
In file included from /home/jocke/openconnect-gui/src/vpninfo.cpp:20:0:
/home/jocke/openconnect-gui/src/vpninfo.h:41:5: error: ‘SOCKET’ does not name a type
     SOCKET get_cmd_fd() const;
     ^
/home/jocke/openconnect-gui/src/vpninfo.h:57:5: error: ‘SOCKET’ does not name a type
     SOCKET cmd_fd;
     ^

These should be easy to fix I think

**Comment by joakim-tjernlund** [comment]
After fixing those I get:
[ 23%] Linking CXX executable ../bin/openconnect-gui
CMakeFiles/openconnect-gui.dir/cert.cpp.o: In function `Cert::clear()':
cert.cpp:(.text+0xbb): undefined reference to `gnutls_x509_crt_deinit'

**Comment by horar** [comment]
Hi @joakim-tjernlund Linux is not primary target platform for his GUI, because there exist other GUI clients integrated into graphical desktops ... but no problem, Branch 'develop' may be compiled now on Linux (Fedora 24, for me).

**Comment by joakim-tjernlund** [comment]
Great! Had a look and I wonder if it is better to include "common.h" in vpninfo.h
rather than defining SOCKET directly

**Comment by horar** [comment]
just a note - yeap, later, that types should be introduced by included library.... so the commot.h may be removed from thiis project.

**Comment by EasyAsABC123** [comment]
anyone get this working?

**Comment by DeonNel1** [comment]
+1
